src/installer.ts

```diff
@@ -241,9 +240,28 @@ export class DotnetCoreInstaller {
     return downloadUrls;
   }
 
-  private getReleasesJson(): Promise<HttpClientResponse> {
-    var httpCallbackClient = new httpClient.HttpClient('setup-dotnet', [], {});
-    return httpCallbackClient.get(DotNetCoreReleasesUrl);
+  private async getReleasesJsonUrl(
+    httpCallbackClient: httpClient.HttpClient,
+    versionParts: string[]
+  ): Promise<string> {
+    // First, get the location of the correct releases.json then get that.
+    const releasesIndex: HttpClientResponse = await httpCallbackClient.get(
+      DotNetCoreIndexUrl
+    );
+    let releasesInfo = JSON.parse(await releasesIndex.readBody())[
+      'releases-index'
+    ];
+    releasesInfo = releasesInfo.filter((releaseInfo: any) => {
+      const sdkParts: string[] = releaseInfo['channel-version'].split('.');
+      if (versionParts.length >= 2 && versionParts[1] != 'x') {
```

Comment:
Lots of magical numbers and values here :) Can we add some comments/descriptive variables? Maybe a sample of the JSON response.